### PR TITLE
Reduce the docker image size

### PR DESCRIPTION
**Pull request by bendavies**
**Summary**

This PR fixes/implements the following **bugs/features**

* [x] Attempt to reduce the size of the final docker image by implementing best practises

The image is large, which takes along time to pull, both locally and in CI.
Reducing the image size would speed up all workflows.

before:
```
thecodingmachine/gotenberg   snapshot          6a331ede6633   58 seconds ago   2.09GB
thecodingmachine/gotenberg   workspace         4e672515939a   35 minutes ago   2.57GB
thecodingmachine/gotenberg   base              0283fe0a2587   4 days ago       2.07GB
```

after:
```
thecodingmachine/gotenberg   snapshot          5fb3376d929d   6 minutes ago    1.07GB
thecodingmachine/gotenberg   workspace         ea1fda548f72   56 minutes ago   1.62GB
thecodingmachine/gotenberg   base              beb19c268974   58 minutes ago   1.05GB
```

**Test plan (required)**

Existing tests should pass

**Checklist**

- [x] Have you followed the guidelines in our [CONTRIBUTING](CONTRIBUTING.md) guide?
- [x] Have you lint your code locally prior to submission (`make lint`)?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally (`make tests`)?
- [x] Have you updated the documentation (Markdown files under `build > docs > content` and then `make doc`)?
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code

### CODE REVIEW

**Comment by gulien** [comment]
Thanks a lot @bendavies!

(On a side note, I've also reduced drastically the number of layers in the new version - wip).

**Comment by gulien** [comment]
@bendavies 👍 

Capture d’écran 2021-03-31 à 11 21 12

**Comment by bendavies** [comment]
thanks @gulien . is your branch pushed?

**Comment by gulien** [comment]
No it’s not, it’s an entirely (private) new repo in fact 🤫

File: build/base/Dockerfile [hunk 1]
```diff
@@ -6,22 +6,29 @@ FROM debian:buster-slim
...
+    rm -rf /var/lib/apt/lists/*
 
-RUN echo "deb http://httpredir.debian.org/debian/ buster main contrib non-free" > /etc/apt/sources.list &&\
-    apt-get update &&\
-    apt-get install -y curl wget gnupg ttf-mscorefonts-installer procps
+# |--------------------------------------------------------------------------
+# | Microsoft font installer
+# |--------------------------------------------------------------------------
+RUN apt-get update &&\
+    curl -o ./ttf-mscorefonts-installer_3.8_all.deb http://httpredir.debian.org/debian/pool/contrib/m/msttcorefonts/ttf-mscorefonts-installer_3.8_all.deb &&\
```
Comment (j2L4e) [inline build/base/Dockerfile:17]: Is it safe to not get the deb via https?

Comment (gulien) [reply build/base/Dockerfile:17]: Looks like a typo, maybe it redirects to https?

**Pull request merged**